### PR TITLE
Add CMake to Detect Scala

### PR DESCRIPTION
**Pull request by abhinavrai44**

### CODE REVIEW

File: cmake/FindScala.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
+set(_SCALA_PATHS
```
Comment (karlnapf) [inline cmake/FindScala.cmake]: Did you write this file? Or did you find it somewhere online? If so, please put a comment with the link.

If you wrote it, double check existing solutions for inspiration

**Comment by karlnapf** [comment]
@vigsterkr I leave merging to you

File: cmake/FindScala.cmake (hunk 1 above)
Comment (abhinavrai44) [inline cmake/FindScala.cmake]: https://github.com/frs69wq/simgrid/blob/master/buildtools/Cmake/Modules/FindScala.cmake

Comment (karlnapf) [inline cmake/FindScala.cmake]: Can you put something like "inspired by [link]" as a comment in this patch?

**Comment by karlnapf** [comment]
Cool so this can be merged.
Did you *test* it locally? I.e. did you run the scala executable with the detected path?

**Comment by abhinavrai44** [comment]
Well, it detect the required files on my system

File: cmake/FindScala.cmake (hunk 1 above)
Comment (abhinavrai44) [inline cmake/FindScala.cmake]: OK

Comment (karlnapf) [inline cmake/FindScala.cmake]: so?

File: cmake/FindScala.cmake [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  /sw
+  /usr
+  /usr/share/java
+  )
+
+find_package(Java COMPONENTS Runtime)
+if(JAVA_FOUND)
+    include(UseJava)
+else()
+    message(WARNING "JAVA Not Found")
```
Comment (karlnapf) [inline cmake/FindScala.cmake]: This needs to be a better message. Imagine the confusion caused by this one :)

File: cmake/FindScala.cmake [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+find_program(Scala_SCALA_EXECUTABLE
+  NAMES scala
+  PATHS ${_SCALA_PATHS}
+  )
+
+if(Scala_SCALA_EXECUTABLE)
+    execute_process(COMMAND ${Scala_SCALA_EXECUTABLE} -version
+      RESULT_VARIABLE res)
+    if( res )
+      message( FATAL_ERROR "Error executing scala -version" )
```
Comment (karlnapf) [inline cmake/FindScala.cmake]: Same here, please provide a better error message

**Comment by karlnapf** [comment]
Good to go apart from minor things ...
@vigsterkr ?

**Comment by vigsterkr** [comment]
missing crucial parts... explained what is needed, waiting for fix.

File: cmake/FindScala.cmake (hunk 1 above)
Comment (abhinavrai44) [inline cmake/FindScala.cmake]: The message got edited in the update. Will put it back in

**Comment by vigsterkr** [comment]
thnx @abhinavrai44 !
could you now rebase your #3597 with the latest develop, since that way we'll see if all the things fit together on travis ;)

**Pull request merged**